From: Ingo Molnar <mingo@elte.hu>
commit 180b65df7ba1e700e28aabfbddbad84b7beebe4b
Author: Ingo Molnar <mingo@elte.hu>
Date:   Tue Nov 25 16:51:45 2008 -0800
Subject: fix warning in fs/dlm/netlink.c
    
    this warning:
    
      fs/dlm/netlink.c: In function ‘dlm_timeout_warn’:
      fs/dlm/netlink.c:131: warning: ‘send_skb’ may be used uninitialized in this function
    
    triggers because GCC does not recognize the (correct) error flow
    between prepare_data() and send_skb.
    
    Annotate it.
    
Signed-off-by: Ingo Molnar <mingo@elte.hu>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Coly Li <coly.li@suse.de>

diff --git a/fs/dlm/netlink.c b/fs/dlm/netlink.c
index 18bda83..aa2a577 100644
--- a/fs/dlm/netlink.c
+++ b/fs/dlm/netlink.c
@@ -127,8 +127,8 @@ static void fill_data(struct dlm_lock_data *data, struct dlm_lkb *lkb)
 
 void dlm_timeout_warn(struct dlm_lkb *lkb)
 {
+	struct sk_buff *uninitialized_var(send_skb);
 	struct dlm_lock_data *data;
-	struct sk_buff *send_skb;
 	size_t size;
 	int rv;
 
